### PR TITLE
Nborovesnkiy/edxoldmng 218/copy certificate configurations including signatures to credentials on certificate update

### PR DESCRIPTION
**Pull request by NikolayBorovenskiy**
**Description:** This PR is going to implement the feature Copy certificate configurations including signatures to credentials on certificate update

**Youtrack:** [Link to Youtrack ticket](https://youtrack.raccoongang.com/issue/EDXOLDMNG-218)

**Configuration instructions:** The https://github.com/raccoongang/openedx-events/pull/1 has to be merged first.

**Reviewers:**
- [ ] @GlugovGrGlib 
- [x] @UvgenGen 

**Merge checklist:**
- [ ] All reviewers approved
- [ ] CI build is green
- [ ] Documentation in source code updated
- [ ] All related Confluence documentation is updated (including deployment documentation)
- [ ] Commits are squashed

**Post merge:**
- [ ] Delete working branch

### CODE REVIEW

File: cms/djangoapps/contentstore/signals/handlers.py [hunk 1]
```diff
@@ -162,6 +239,56 @@ def listen_for_course_catalog_info_changed(sender, signal, **kwargs):
...
+    """
+    certificate_config = kwargs['certificate_config']
+    try:
+        credentials_client = get_credentials_api_client(
+            User.objects.get(username=settings.CREDENTIALS_SERVICE_USERNAME),
+        )
+        credentials_api_base_url = get_credentials_api_base_url()
+        api_url = urljoin(f'{credentials_api_base_url}/', 'course_certificates/')
+        payload = attr.asdict(certificate_config)
+        payload['course_key'] = str(payload.pop('course_key'))
```
Comment (NiedielnitsevIvan) [inline cms/djangoapps/contentstore/signals/handlers.py:281]: And here too.

File: cms/djangoapps/contentstore/signals/handlers.py [hunk 2]
```diff
@@ -162,6 +239,56 @@ def listen_for_course_catalog_info_changed(sender, signal, **kwargs):
...
+    certificate_config = kwargs['certificate_config']
+    files_to_upload = dict(get_certificate_signature_assets(certificate_config))
+    try:
+        credentials_client = get_credentials_api_client(
+            User.objects.get(username=settings.CREDENTIALS_SERVICE_USERNAME),
+        )
+        credentials_api_base_url = get_credentials_api_base_url()
+        api_url = urljoin(f'{credentials_api_base_url}/', 'course_certificates/')
+        payload = attr.asdict(certificate_config)
+        payload['course_key'] = str(payload.pop('course_key'))
```
Comment (NiedielnitsevIvan) [inline cms/djangoapps/contentstore/signals/handlers.py:256]: Change it to `course_id` pls. Because the API takes the course_id argument by default.

Comment (NikolayBorovenskiy) [reply cms/djangoapps/contentstore/signals/handlers.py:256]: Agree and I've fixed it already here https://github.com/raccoongang/edx-platform/pull/2465/files. Please look it also. This PR is going to be merged together with the current one.

File: cms/djangoapps/contentstore/signals/handlers.py (hunk 1 above)
Comment (NikolayBorovenskiy) [reply cms/djangoapps/contentstore/signals/handlers.py:281]: Yep, fixed here https://github.com/raccoongang/edx-platform/pull/2465/files.

File: cms/djangoapps/contentstore/signals/handlers.py (hunk 2 above)
Comment (NiedielnitsevIvan) [reply cms/djangoapps/contentstore/signals/handlers.py:256]: It's ok that you pass `course_key` to CertificateConfigData here, but in RP https://github.com/raccoongang/edx-platform/pull/2465/files you pass `course_id`?

Comment (NikolayBorovenskiy) [reply cms/djangoapps/contentstore/signals/handlers.py:256]: Do not understand the question a little bit. There is no already `course_key` at all. Only we get now `course_id`. I've changed it everwhere. Please explain what do u mean.)

**Comment by UvgenGen** [comment]
@NikolayBorovenskiy Main case for failed tests is `ImportError: cannot import name 'CertificateConfigData' from 'openedx_events.content_authoring.data'`. Also take a look at linter errors, please.

**Pull request merged**